dataproc/python-api-walkthrough.md

```diff
@@ -0,0 +1,158 @@
+
+## Prerequisites (2)
+
+1. This walkthrough uploads a PySpark file (`pyspark_sort.py`) to a
+   [Cloud Storage bucket](https://cloud.google.com/storage/docs/key-terms#buckets) in
+   your project.
+   * You can use the [Cloud Storage browser page](https://console.cloud.google.com/storage/browser)
+   in Google Cloud Platform Console to view existing buckets in your project.
+   * To create a new bucket, run the following command. Your bucket name must be unique.
+   ```bash
+   gsutil mb -p {{project-id}} gs://your-bucket-name
+   ```
+
+1.  Set environment variables.
+
+    * Set the name of your bucket.
+    ```bash
+    BUCKET=your-bucket-name
+    ```
+
+## Prerequisites (3)
+
+1. Set up a Python
+   [virtual environment](https://virtualenv.readthedocs.org/en/latest/)
+   in Cloud Shell.
+
+    * Create the virtual environment.
+    ```bash
+    virtualenv ENV
+    ```
+    * Activate the virtual environment.
+    ```bash
+    source ENV/bin/activate
+    ```
+
+1. Install library dependencies in Cloud Shell.
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+## Create a cluster and submit a job
+
+1. Set a name for your new cluster.
+    ```bash
+    CLUSTER=new-cluster-name
+    ```
+
+1. Set a [zone](https://cloud.google.com/compute/docs/regions-zones/#available)
+   where your new cluster will be located. You can change the
+   "us-central1-a" zone that is pre-set in the following command.
+    ```bash
+    ZONE=us-central1-a
+    ```
+
+1. Run `submit_job.py` with the `--create_new_cluster` flag
+   to create a new cluster and submit the `pyspark_sort.py` job
+   to the cluster.
+    ```bash
+    python submit_job_to_cluster.py \
+    --project_id={{project-id}} \
+    --cluster_name=$CLUSTER \
+    --zone=$ZONE \
+    --gcs_bucket=$BUCKET \
+    --create_new_cluster
+    ```
+
+## Job Output
+
+Job output in Cloud Shell shows cluster creation, job submission,
+    job completion, and then tear-down of the cluster.
+
+     ```bash
+     ...
+     Creating cluster...
+     Cluster created.
+     Uploading pyspark file to GCS
+     new-cluster-name - RUNNING
+     Submitted job ID ...
+     Waiting for job to finish...
+     Job finished.
+     Downloading output file
+     .....
+     ['Hello,', 'dog', 'elephant', 'panther', 'world!']
+     ...
+     Tearing down cluster
+     ```
+## Congratulations on Completing the Walkthrough!
+<walkthrough-conclusion-trophy></walkthrough-conclusion-trophy>
+
+---
+
+### Next Steps:
+
+* **View job details from the Console.** View job details by selecting the
+   PySpark job from the Cloud Dataproc 
+   [Jobs page](https://console.cloud.google.com/dataproc/jobs)
+   in the Google Cloud Platform Console.
+
+* **Delete resources used in the walkthrough.**
+   The `submit_job.py` job deletes the cluster that it created for this
+   walkthrough. You can run the following command to delete the
+   Cloud Storage bucket used in this walkthrough (the bucket must be empty).
```

Comment:
Can we suggest deleting the bucket only if they created one for the tutorial? It wouldn't be good to accidentally delete an existing bucket.

Reply:
Done.